login/service_blackbox_test.go

```diff
@@ -510,26 +450,43 @@ func (s *serviceBlackBoxTest) TestInvalidOAuthAuthorizationCode() {
 	assert.Contains(s.T(), locationString, refererUrl)
 }
 
+func (s *serviceBlackBoxTest) getDummyOauthIDPService(forApprovedUser bool) *dummyIDPOauthService {
+	g := s.NewTestGraph(s.T())
+	newIdentity := g.CreateUser().Identity()
+	claims := make(map[string]interface{})
+	claims["sub"] = uuid.NewV4()
+	if forApprovedUser {
+		claims["preferred_username"] = newIdentity.Username
+		claims["email"] = newIdentity.User.Email
+		claims["company"] = newIdentity.User.Company
+	}
+	accessToken, err := testtoken.GenerateTokenWithClaims(claims)
+	require.Nil(s.T(), err)
```

Comment:
`require.NoError(s.T(), err)` is better than `require.Nil(..)` because it will display the error message if something wrong happened